### ISSUE TITLE
Add Pre and Post Build Scripts

### ISSUE THREAD

**Issue by ThadHouse** [robotdotnet/WPILib#57]
I don't like manually handling the native libraries. Would be nice to have a pre-build script to automatically download a specified native library.

~~Would also be nice to have a post-build script to automatically use sandcastle to generate the proper XML.~~

**Comment by ThadHouse** [comment]
The sandcastle portion of this is done.

**Comment by ThadHouse** [comment]
Pre build executable is written. Still needs to be added to the project, but I want to wait on that until we have more then just the athena hal library.

**Comment by ThadHouse** [comment]
PreBuild script is done
